packages/ui/src/composables/useBem.ts

```diff
@@ -1,21 +1,59 @@
-import { computed, Ref, unref } from 'vue'
+import { computed, Ref, unref, ComputedRef } from 'vue'
+import { __DEV__ } from '../utils/global-utils'
 
-export const useBem = (baseClass: string) => {
-  const createModifiersClasses = (modifiersCb: () => Record<string, boolean>, classes: string[] | Ref<string[]> = []) => {
-    return computed(() => {
-      const modifiers = modifiersCb()
-      return [
-        baseClass,
-        ...unref(classes),
-        ...Object
-          .keys(modifiers)
-          .filter((modifierName) => modifiers[modifierName])
-          .map((modifierName) => `${baseClass}--${modifierName}`),
-      ]
-    })
-  }
+type ComputedClasses = Ref<Record<string, any>> & {
+  value: ComputedRef<Record<string, true>>
+  asObject: ComputedRef<Record<string, true>>
+  asArray: ComputedRef<string[]>
+  asString: ComputedRef<string>
+}
 
-  return {
-    createModifiersClasses,
+/**
+ * @description creates BEM modifiers based on transferred prefix (base BEM class) & modifiers list.
+ * @param prefix string that classes start with (base BEM class).
+ * @param modifiers list of options that will serve as state BEM modifiers.
+ * @returns computed classes starting with "prefix" and ending with form state BEM modifier.
+ * @example
+ *  const result = useBem('va-component', computed(() => pick(props, ['success'])))
+ *  // if success is `true`
+ *  result.value.value: { 'va-component--success': true }
+ *  result.asObject.value: { 'va-component--success': true }
+ *  result.asArray.value: ['va-component--success']
+ *  result.asString.value: 'va-component--success'
+ */
+export const useBem = (
+  prefix = '',
+  modifiers: Record<string, boolean> | Ref<Record<string, boolean>>,
+) => {
+  if (__DEV__ && !prefix) {
+    console.warn('You must pass the @param "prefix" to the useBem hook!')
   }
+
+  const computedBemClassesObject = computed(() => {
+    return Object
+      .entries(unref(modifiers))
+      .reduce((classesObj: Record<string, boolean>, [modifiersName, value]) => {
+        if (value) { classesObj[`${prefix}--${modifiersName}`] = true }
+        return classesObj
+      }, {})
+  })
+
+  const computedBemClassesArray = computed(() => Object.keys(computedBemClassesObject.value))
+
+  const computedBemClassesString = computed(() => computedBemClassesArray.value.join(' '))
+
+  return new Proxy({}, {
+    get (_, key: string) {
+      switch (key) {
+        case 'asArray':
+          return computedBemClassesArray
+        case 'asString':
+          return computedBemClassesString
+        case 'asObject':
+          return computedBemClassesObject
+        default:
+          return computedBemClassesObject
```

Comment:
```suggestion
          return computedBemClassesObject.value[key]
```

Do I understand it correctly?